lib/WTSI/NPG/iRODS/Publisher.pm

```diff
@@ -552,21 +553,27 @@ sub _read_md5_cache_file {
 sub _make_md5_cache_file {
   my ($self, $cache_file, $md5) = @_;
 
-  $self->warn("Adding missing MD5 cache file '$cache_file'");
-
-  try {
-    my $out;
-    open $out, '>', $cache_file or
-      croak "Failed to open '$cache_file' for writing: $ERRNO";
-    print $out "$md5\n" or
-      croak "Failed to write MD5 to '$cache_file': $ERRNO";
-    close $out or
-      $self->warn("Failed to close '$cache_file' cleanly: $ERRNO");
-  } catch {
-    # Failure to create a cache should not be a hard error. Here we
-    # just forward the message from croak above.
-    $self->warn($_);
-  };
+  $self->debug("Adding missing MD5 cache file '$cache_file'");
+
+  my ($filename, $cache_dir) = fileparse($cache_file);
+  if (-w $cache_dir) {
```

Comment:
we need to test for -x as well since if the directory is writable, but execute permission is denied, we will not be able to create a file inside it

Reply:
I agree - added.